### ISSUE TITLE
Make button's focus outline consistent with its shape

### ISSUE THREAD

**Issue by joaopaulobdac** [uber/baseweb#3647]
## Current Behavior

Using a theme with custom `buttonBorderRadius`: **(baseweb is using the `outline` css property here)**

![Peek 2020-08-17 17-27](https://user-images.githubusercontent.com/62244135/90441262-0a01a500-e0af-11ea-991b-5c6176242822.gif)

When I set the button's shape to `pill` it looks right, but it ends up overriding the `buttonBorderRadius` theme property: **(baseweb is now using `box-shadow` instead)**

![20200817170817](https://user-images.githubusercontent.com/62244135/90440862-4ed90c00-e0ae-11ea-8657-1deec598af57.png)

## Expected Behavior

The focus outline should not ignore `buttonBorderRadius` thus having a consistent shape. Since the `outline` css property doesn't follow `border-radius`, it should be replaced in favor of `box-shadow` even if the button shape is not set to `pill`.

Relevant:

https://github.com/uber/baseweb/blob/e8f95c91498fbbc023a41f29ca6159fd6ca5d5c1/src/button/styled-components.js#L43

## Your Environment

| Tech    | Version |
| ------- | ------- |
| Base UI | v9.95.1 |
| React | v16.13.1 |
| browser | Chromium v84.0.4147.125 |

- [x] I have searched the [issues](https://github.com/uber/baseweb/issues) of this repository and believe that this is not a duplicate.

**Comment by joaopaulobdac** [comment]
Hey @sandgraham, thank you! Do you think it would be feasible to completely replace `outline` in general? I have also encountered this issue with rounded checkboxes now, and I imagine I will again with other focused components as well

**Comment by sandgraham** [comment]
@joaopaulobdac - I think it makes sense to do this for anything that can be rounded via the theme at the very least. I'll make a new issue to track that effort.
